pommerman/agents/player_agent.py

```diff
@@ -25,20 +82,40 @@ def __init__(self, agent, control):
                 key.A: 3,
                 key.D: 4,
                 key.E: 5,
-                key.Q: 6 # In Pommerman, this will freeze the game.
+                # key.Q: 6 # In Pommerman, this will freeze the game.
             }
         else:
             raise ValueError("Unknown player control '{}'".format(control))
 
     def act(self, obs, action_space):
-        return self._key_input['curr']
+        # Go through the keys and fire for those that needs repetition (because they're held down)
+        for k, state in self._keystate.items():
+            if state.fires():
+                self._q.append(k)
+                state.fire()
+
+        act = 0
+        if self._q:  # Work off the keys that are queued.
+            act = self._key2act.get(self._q.pop(0), 0)
```

Comment:
when would this use the default "0"?

Reply:
Legacy; you're right, it can't in the current implementation, so I'll change it to `[]`.